execute/tokendata/observer_background_test.go

```diff
@@ -0,0 +1,146 @@
+package tokendata
+
+import (
+	"fmt"
+	rand2 "math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/testhelpers/rand"
+	"github.com/smartcontractkit/chainlink-ccip/internal/mocks"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+//nolint:gocyclo // this is a test
+func Test_backgroundObserver(t *testing.T) {
```

Comment:
I think we missed some advanced scenarios here, for instance
* multiple tokenDataObservers in baseObserver
  * some faulty, some succesful
  * all faulty
  * all successful